src/main/java/com/github/elmo675/controller/DiaryEntryController.java

```diff
@@ -13,51 +18,119 @@
+        diaryEntry.setUpdatedBy(diaryEntryResponse.getUpdatedBy());
+        diaryEntry.setUpdatedAt(diaryEntryResponse.getUpdatedAt());
+        diaryEntry.setCreatedAt(diaryEntryResponse.getCreatedAt());
+        diaryEntry.setCreatedBy(diaryEntryResponse.getCreatedBy());
+        return diaryEntry;
+    }
+    private DiaryEntry convertRequestToDiaryEntry(DiaryEntryRequest diaryEntryRequest) throws ParseException {
+        DiaryEntry diaryEntry = modelMapper.map(diaryEntryRequest, DiaryEntry.class);
+        //DiaryEntry diaryEntry = new DiaryEntry();
+        diaryEntry.setContent(diaryEntryRequest.getContent());
+        diaryEntry.setAccess(diaryEntryRequest.getAccess());
+        if(diaryEntry.getCreatedBy() != null){
+            diaryEntry.setUpdatedBy(diaryEntryRequest.getCreatedBy());
+            diaryEntry.setUpdatedAt(new Date());
+        }
+        else{
+            Date date = new Date();
+            diaryEntry.setCreatedBy(diaryEntryRequest.getCreatedBy());
+            diaryEntry.setUpdatedBy(diaryEntryRequest.getCreatedBy());
+            diaryEntry.setCreatedAt(date);
+            diaryEntry.setUpdatedAt(date);
+        }
+        return diaryEntry;
+    }
+
+
+
+
+
     @GetMapping("/entry")
-    public List<DiaryEntry> getAllDiaryEntries() {
-        return DiaryEntryRepository.findAll();
+    public List<DiaryEntryResponse> getAllDiaryEntries() {
+        List <DiaryEntry> entryList = DiaryEntryRepository.findAll();
+        return entryList.stream().map(this::convertDiaryEntryToResponse).collect(Collectors.toList());
     }
 
+
     @GetMapping("/entry/{id}")
-    public ResponseEntity<DiaryEntry> getDiaryEntryById(@PathVariable(value = "id") Long DiaryEntryId)
+    public ResponseEntity<DiaryEntryResponse> getDiaryEntryById(@PathVariable(value = "id") Long diaryEntryId)
             throws ResourceNotFoundException {
-        DiaryEntry DiaryEntry =
+        DiaryEntry diaryEntry =
                 DiaryEntryRepository
-                        .findById(DiaryEntryId)
-                        .orElseThrow(() -> new ResourceNotFoundException("DiaryEntry not found on :: " + DiaryEntryId));
-        return ResponseEntity.ok().body(DiaryEntry);
+                        .findById(diaryEntryId)
+                        .orElseThrow(() -> new ResourceNotFoundException("DiaryEntry not found on :: " + diaryEntryId));
+        return ResponseEntity.ok().body(convertDiaryEntryToResponse(diaryEntry));
     }
+
     @PostMapping("/entry")
-    public DiaryEntry createDiaryEntry(@Valid @RequestBody DiaryEntry DiaryEntry) {
-        return DiaryEntryRepository.save(DiaryEntry);
+    public DiaryEntryResponse createDiaryEntry(@Valid @RequestBody DiaryEntryRequest diaryEntryRequest) {
+        try {
+            DiaryEntry result = convertRequestToDiaryEntry(diaryEntryRequest);
+            return convertDiaryEntryToResponse(DiaryEntryRepository.save(result));
+        }
+        catch (ParseException e){
```

Comment:
1. catch robi się w tej samej linii co klamra
2. kiedy leci ten `ParseException`? zwrócenie `null` w takiej sytuacji to dobre podejście?